cl/beacon/handler/block_production.go

```diff
@@ -320,7 +424,113 @@ func (a *ApiHandler) produceBeaconBody(ctx context.Context, apiVersion int, base
 	return beaconBody, executionValue, nil
 }
 
-func (a *ApiHandler) setupHeaderReponseForBlockProduction(w http.ResponseWriter, consensusVersion clparams.StateVersion, blinded bool, executionBlockValue, consensusBlockValue uint64) {
+func (a *ApiHandler) getBlockOperations(s *state.CachingBeaconState, targetSlot uint64) (
+	*solid.ListSSZ[*cltypes.AttesterSlashing],
+	*solid.ListSSZ[*cltypes.ProposerSlashing],
+	*solid.ListSSZ[*cltypes.SignedVoluntaryExit],
+	*solid.ListSSZ[*cltypes.SignedBLSToExecutionChange]) {
+
+	attesterSlashings := solid.NewDynamicListSSZ[*cltypes.AttesterSlashing](
+		int(a.beaconChainCfg.MaxAttesterSlashings),
+	)
+	slashedIndicies := []uint64{}
+	// AttesterSlashings
+AttLoop:
+	for _, slashing := range a.operationsPool.AttesterSlashingsPool.Raw() {
+		idxs := slashing.Attestation_1.AttestingIndices
+		rawIdxs := []uint64{}
+		for i := 0; i < idxs.Length(); i++ {
+			currentValidatorIndex := idxs.Get(i)
+			if slices.Contains(slashedIndicies, currentValidatorIndex) || slices.Contains(rawIdxs, currentValidatorIndex) {
+				continue AttLoop
+			}
+			v := s.ValidatorSet().Get(int(currentValidatorIndex))
+			if !v.IsSlashable(targetSlot / a.beaconChainCfg.SlotsPerEpoch) {
+				continue AttLoop
+			}
+			rawIdxs = append(rawIdxs, currentValidatorIndex)
+		}
+		slashedIndicies = append(slashedIndicies, rawIdxs...)
+		attesterSlashings.Append(slashing)
+		if attesterSlashings.Len() >= int(a.beaconChainCfg.MaxAttesterSlashings) {
+			break
+		}
+	}
+	// ProposerSlashings
+	proposerSlashings := solid.NewStaticListSSZ[*cltypes.ProposerSlashing](
+		int(a.beaconChainCfg.MaxProposerSlashings),
+		416,
```

Comment:
could be better if can define this bytes len as a static constant somewhere else.